pkg/controller/registry/resolver/resolver.go

```diff
@@ -548,20 +548,32 @@ func (r *SatResolver) sortChannel(bundles []*Operator) ([]*Operator, error) {
 		}
 	}
 
-	if len(headCandidates) != 1 {
-		// TODO: more context in error
-		return nil, fmt.Errorf("found more than one head for channel")
+	if len(headCandidates) > 1 {
+		var p string
+		for _, v := range headCandidates {
+			if v.sourceInfo.Package != p {
+				p = v.sourceInfo.Package
+			} else {
+				// if find multi heads in the same package, return error
+				return nil, fmt.Errorf("found more than one head for package: %s, please check the `replaces` or `skipRange` fields of the operator bundles", p)
+			}
+		}
+
+	} else if len(headCandidates) < 1 {
+		return nil, fmt.Errorf("Not found the head for channel")
 	}
 
-	head := headCandidates[0]
-	current := head
-	for {
-		channel = append(channel, current)
-		next, ok := replaces[current]
-		if !ok {
-			break
+	for _, v := range headCandidates {
```

Comment:
Is this change just for the case where `headCandidates == nil`? Is it necessary since above we are just already returning if that's the case?

Reply:
Not really, this change is for the situation of the `len(headCandidates) > 1`.  If there are multi packages in a channel, there will multi `headCandidates`, return all of them here.